json_encoder.go

```diff
@@ -141,6 +141,12 @@ func (enc *jsonEncoder) AddUint64(key string, val uint64) {
 	enc.bytes = strconv.AppendUint(enc.bytes, val, 10)
 }
 
+func (enc *jsonEncoder) AddUintptr(key string, val uintptr) {
+	enc.addKey(key)
+	enc.bytes = append(enc.bytes, '0', 'x')
+	enc.bytes = strconv.AppendUint(enc.bytes, uint64(val), 16)
```

Comment:
Thank you for pointing out the JSON spec. :) I have also removed support for `unsafe.Pointer` and squashed the commits.